pkg/steps/new_e2e_test.go

```diff
@@ -13,10 +13,27 @@ import (
 	"github.com/openshift/ci-tools/pkg/api"
 )
 
+type testParameters struct {
+	m map[string]string
+}
+
+func (p testParameters) Has(name string) bool {
+	_, ok := p.m[name]
+	return ok
+}
+func (p testParameters) HasInput(name string) bool        { return false }
+func (p testParameters) Get(name string) (string, error)  { return p.m[name], nil }
+func (p testParameters) Links(name string) []api.StepLink { return nil }
+
 func TestGenPods(t *testing.T) {
 	config := api.TestStepConfiguration{
 		As: "test",
 	}
+	params := testParameters{
+		m: map[string]string{
+			"IMAGE_IMAGE0": "registry.example.com/test/image0",
```

Comment:
Why didn't we see this affect the expected test output?

Reply:
Because `steps[0].Image` should have been just `image0`. Bad test.